cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -148,14 +148,15 @@ def __post_init__(self):
                     raise ValueError('repetitions are negative but the circuit is not invertible')
 
             # Initialize repetition_ids to default, if unspecified. Else, validate their length.
-            loop_size = abs(self.repetitions)
-            if not self.repetition_ids:
-                object.__setattr__(self, 'repetition_ids', self._default_repetition_ids())
-            elif len(self.repetition_ids) != loop_size:
-                raise ValueError(
-                    f'Expected repetition_ids to be a list of length {loop_size}, '
-                    f'got: {self.repetition_ids}'
-                )
+            if self.use_repetition_ids:
+                loop_size = abs(self.repetitions)
+                if not self.repetition_ids:
+                    object.__setattr__(self, 'repetition_ids', self._default_repetition_ids())
```

Comment:
Could all these be consolidated by having `default_repetition_ids` return None if `use_repetition_ids ` is false?

Reply:
Changed to do the `self.use_repetition_ids` check in `self._default_repetition_ids`.